zipline/finance/performance/position.py

```diff
@@ -172,13 +172,13 @@ def adjust_commission_cost_basis(self, asset, cost):
         if self.amount == 0:
             return
 
-        prev_cost = self.cost_basis * self.amount
+        prev_cost = self.cost_basis * abs(self.amount)
```

Comment:
@ssanderson and I had an in person chat were we discussed whether this was the correct change or not.

That conversation hinges on what we determine on what we intend `cost_basis` to mean. If it is meant to be a value to represent a price at which the the position would break even, the current implementation was correct. 

However, if cost basis is meant as the cost basis for tax reporting purposes, then I believe your change is correct. Since, during a short sale, the stocks are briefly held and then sold at current market price. I can definitely be wrong here, and we should verify, but in that case for tax reasons the cost basis would be from that sale of the shares.

(I checked `Position.update`, and there I do believe we are correctly handling the `cost_basis` for the second definition, when covering a short, by holding on to the `cost_basis` of the initial short sale until the short is fully covered.)

If that clarification blocks, we could separate this commit out from the per units fix.